Tests/SwiftParserTest/translated/MatchingPatternsTests.swift

```diff
@@ -606,4 +606,102 @@ final class MatchingPatternsTests: ParserTestCase {
       experimentalFeatures: .referenceBindings
     )
   }
+
+  func testBorrowingContextualParsing() {
+    assertParse(
+      """
+      switch bar {
+      case _borrowing .foo(): // parses as `_borrowing.foo()` as before
+        break
+      }
+      """,
+      experimentalFeatures: .borrowingSwitch
+    )
+
+    assertParse(
+      """
+      switch bar {
+      case _borrowing (): // parses as `_borrowing()` as before
+        break
+      }
+      """,
+      experimentalFeatures: .borrowingSwitch
+    )
+
+    assertParse(
+      """
+      switch bar {
+      case _borrowing x: // parses as binding
+        break
+      }
+      """,
+      experimentalFeatures: .borrowingSwitch
+    )
```

Comment:
You can check that it actually parses as a binding (and in the previous test case that it parses as a pattern) by adding a `substructure` parameter. The easiest way to generate them is to run a single `assertParse` function, set a breakpoint in `assertParse`, run `po tree.debugInitCall(includeTrivia: false)`, trim it down and then add it as the `substructure` parameter to `assertParse`.